### PR TITLE
Obstacle course fourth

### PR DESCRIPTION
**Pull request by k-mma**
My code turns 90 degrees to the right, moves forward in a diagonal line to in front of the truss, and moves forward in a straight line under the truss.
Start point (-9, 59, 0°)
End point (-5, -30, 270°)
Screenshot 2024-09-04 163847

Updated path with all parts except first
Screenshot 2024-09-06 164603

### CODE REVIEW

**Review by CharlesJfkRbt — APPROVED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/obstacleCoursePathFour.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.firstinspires.ftc.teamcode.drive.SampleMecanumDrive;
+import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequence;
+
+@Autonomous (group = "drive")
+
+public class obstacleCoursePathFour extends LinearOpMode {
+
+    @Override
+
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/obstacleCoursePathFour.java]: Put the annotation directly above the function

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/obstacleCoursePathFour.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.acmerobotics.roadrunner.geometry.Pose2d;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+
+import org.firstinspires.ftc.teamcode.drive.SampleMecanumDrive;
+import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequence;
+
+@Autonomous (group = "drive")
+
+public class obstacleCoursePathFour extends LinearOpMode {
```
Comment (CubeOfCheese) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/obstacleCoursePathFour.java]: put annotation directly above the class

**Review by CubeOfCheese — CHANGES_REQUESTED** [review]

**Comment by AkithMadurapperuma** [comment]
My end point coordinates are (-10, 55) but your starting point in this code is (-9, 59). You should probably update your code since my chunk is before this one.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 3]
```diff
@@ -31,7 +31,12 @@ public void runOpMode() {
         Trajectory backToStart = drive.trajectoryBuilder(new Pose2d(48, 55, Math.toRadians(0)))
                 .lineTo(new Vector2d(-10, 55))
                 .build();
-
+        // fourth part
+        TrajectorySequence driveUnderObstacle = drive.trajectorySequenceBuilder(new Pose2d(-10, 55, Math.toRadians(0)))
+                .turn(Math.toRadians(-90))
+                .lineToSplineHeading(new Pose2d(-5, 30, Math.toRadians(270)))
```
Comment (isabellammjfk) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java:37]: is there meant to be a duplicate??

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java [hunk 4]
```diff
@@ -55,10 +60,12 @@ public void runOpMode() {
         drive.turn(Math.toRadians(-90));
         drive.followTrajectorySequence(forwardToWoodBack);
         drive.followTrajectory(backToStart);
+        // fourth part
```
Comment (isabellammjfk) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/drive/opmode/ObstacleCoursePath.java:63]: you should also have comments to differentiate between when each part is being called at the bottom

**Review by isabellammjfk — COMMENTED** [review]
lowk a dub but keeping it real and good job queen

**Review by CubeOfCheese — APPROVED** [review]

**Review by isabellammjfk — APPROVED** [review]
muy biennnnnnnnnnnn

**Pull request merged**